piet-gpu/src/render_ctx.rs

```diff
@@ -303,6 +303,14 @@ impl PietGpuRenderContext {
     }
 
     fn encode_path(&mut self, path: impl Iterator<Item = PathEl>, is_fill: bool) {
+        if is_fill {
+            self.encode_path_inner(path.chain(std::iter::once(PathEl::ClosePath)), is_fill)
```

Comment:
This is also not quite right (and makes me realize a similar issue in the original version) - if there are multiple open subpaths, each one needs an implicit close, not just at the end. Sorry for not communicating this earlier!

Reply:
Good catch! Working on this.